[Bug]: Build failure with llvm clang-19

### Before proceeding, is there an existing issue or discussion for this?

- [x] I have done a search for similar [issues](.?q=type%3Aissue) and [discussions](https://discourse.openrobotics.org/c/open-rmf/open-rmf-general/103).

### OS and version

Ubuntu 24.04

### Open-RMF installation type

Source build

### Other Open-RMF installation methods

_No response_

### Open-RMF version or commit hash

main

### ROS distribution

Jazzy

### ROS installation type

Binaries

### Other ROS installation methods

_No response_

### Package or library, if applicable

_No response_

### Description of the bug

It seems like llvm clang-19 is having issues with the `on_error_notification` strcut in `rxcpp`. It seems to be correctly calling out that the member variable `const rxu::error_ptr ep;` which is marked `const` is being `moved` in one of the operator overloads?

### Steps to reproduce the bug

sudo apt install clang-19
CC=/usr/bin/clang-19 CXX=/usr/bin/clang++-19 colcon build --packages-up-to rmf_fleet_adapter

### Expected behavior

Build succeeds

### Actual behavior

```
In file included from /home/yadunund/ws_rmf/src/rmf/rmf_ros2/rmf_fleet_adapter/src/rmf_fleet_adapter/services/FindEmergencyPullover.cpp:18:
In file included from /home/yadunund/ws_rmf/src/rmf/rmf_ros2/rmf_fleet_adapter/src/rmf_fleet_adapter/services/FindEmergencyPullover.hpp:21:
In file included from /home/yadunund/ws_rmf/src/rmf/rmf_ros2/rmf_fleet_adapter/src/rmf_fleet_adapter/services/../jobs/SearchForPath.hpp:21:
In file included from /home/yadunund/ws_rmf/src/rmf/rmf_ros2/rmf_fleet_adapter/src/rmf_fleet_adapter/services/../jobs/Planning.hpp:21:
In file included from /home/yadunund/ws_rmf/src/rmf/rmf_ros2/rmf_fleet_adapter/rmf_rxcpp/include/rmf_rxcpp/RxJobs.hpp:21:
In file included from /home/yadunund/ws_rmf/src/rmf/rmf_ros2/rmf_fleet_adapter/rmf_rxcpp/include/rmf_rxcpp/detail/RxJobsDetail.hpp:21:
In file included from /home/yadunund/ws_rmf/src/rmf/rmf_ros2/rmf_fleet_adapter/rmf_rxcpp/RxCpp-4.1.0/Rx/v2/src/rxcpp/rx.hpp:8:
In file included from /home/yadunund/ws_rmf/src/rmf/rmf_ros2/rmf_fleet_adapter/rmf_rxcpp/RxCpp-4.1.0/Rx/v2/src/rxcpp/rx-includes.hpp:208:
/home/yadunund/ws_rmf/src/rmf/rmf_ros2/rmf_fleet_adapter/rmf_rxcpp/RxCpp-4.1.0/Rx/v2/src/rxcpp/rx-notification.hpp:153:72: error: no viable overloaded '='
  153 |         on_error_notification& operator=(on_error_notification o) { ep = std::move(o.ep); return *this; }
      |                                                                     ~~ ^ ~~~~~~~~~~~~~~~
/usr/lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/exception_ptr.h:213:20: note: candidate function not viable: 'this' argument has type 'const rxu::error_ptr' (aka 'const std::__exception_ptr::exception_ptr'), but method is not marked const
  213 |     exception_ptr::operator=(const exception_ptr& __other) _GLIBCXX_USE_NOEXCEPT
      |                    ^
/usr/lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/exception_ptr.h:140:7: note: candidate function not viable: 'this' argument has type 'const rxu::error_ptr' (aka 'const std::__exception_ptr::exception_ptr'), but method is not marked const
  140 |       operator=(exception_ptr&& __o) noexcept
```

### Additional information or screenshots

_No response_

--- a/rmf_fleet_adapter/rmf_rxcpp/RxCpp-4.1.0/Rx/v2/src/rxcpp/rx-notification.hpp
+++ b/rmf_fleet_adapter/rmf_rxcpp/RxCpp-4.1.0/Rx/v2/src/rxcpp/rx-notification.hpp
@@ -146,11 +146,12 @@ private:
     };
 
     struct on_error_notification : public base {
-        on_error_notification(rxu::error_ptr ep) : ep(ep) {
-        }
-        on_error_notification(const on_error_notification& o) : ep(o.ep) {}
-        on_error_notification(const on_error_notification&& o) : ep(std::move(o.ep)) {}
-        on_error_notification& operator=(on_error_notification o) { ep = std::move(o.ep); return *this; }
+        explicit on_error_notification(rxu::error_ptr ep_arg) : ep(std::move(ep_arg)) {}
+        virtual ~on_error_notification() = default;
+        on_error_notification(const on_error_notification& other) = default;
+        on_error_notification& operator=(const on_error_notification& other) = default;
+        on_error_notification(on_error_notification&& other) noexcept = default;
+        on_error_notification& operator=(on_error_notification&& other) noexcept = default;
         virtual void out(std::ostream& os) const {
             os << "on_error(";
             os << rxu::what(ep);
@@ -167,7 +168,7 @@ private:
         virtual void accept(const typename base::observer_type& o) const {
             o.on_error(ep);
         }
-        const rxu::error_ptr ep;
+        rxu::error_ptr ep;
     };
 
     struct on_completed_notification : public base {
@@ -264,7 +265,7 @@ std::ostream& operator<< (std::ostream& out, const recorded<T>& r) {
     out << "@" << r.time() << "-" << r.value();
     return out;
 }
- 
+
 }
 namespace rxn=notifications;
 

